cmd/app/add.go

```diff
@@ -85,38 +85,36 @@ func preRunAddCommand(ctx context.Context, clients *shared.ClientFactory) error
 	if !clients.Config.WithExperimentOn(experiment.BoltFrameworks) {
 		return nil
 	}
-	manifestSource, err := clients.Config.ProjectConfig.GetManifestSource(ctx)
-	if err != nil {
-		return err
-	}
-	if manifestSource.Equals(config.ManifestSourceRemote) {
-		return slackerror.New(slackerror.ErrAppInstall).
-			WithMessage("Apps cannot be installed due to project configurations").
-			WithRemediation(
-				"Install an app on app settings: %s\nLink an app to this project with %s\nList apps saved with this project using %s",
-				style.LinkText("https://api.slack.com/apps"),
-				style.Commandf("app link", false),
-				style.Commandf("app list", false),
-			).
-			WithDetails(slackerror.ErrorDetails{
-				slackerror.ErrorDetail{
-					Code:    slackerror.ErrProjectConfigManifestSource,
-					Message: "Cannot install apps with manifests sourced from app settings",
-				},
-			})
-	}
 	return nil
 }
 
 // RunAddCommand executes the workspace install command, prints output, and returns any errors.
 func RunAddCommand(ctx context.Context, clients *shared.ClientFactory, selection *prompts.SelectedApp, orgGrantWorkspaceID string) (context.Context, types.InstallState, types.App, error) {
 	if selection == nil {
-		selected, err := teamAppSelectPromptFunc(ctx, clients, prompts.ShowHostedOnly, prompts.ShowAllApps)
+		// Prompt for deployed or local app environment.
+		isProductionApp, err := promptIsProduction(ctx, clients)
+		if err != nil {
+			return ctx, "", types.App{}, err
+		}
+
+		var appEnvironmentType prompts.AppEnvironmentType
+		if isProductionApp {
+			appEnvironmentType = prompts.ShowHostedOnly
+		} else {
+			appEnvironmentType = prompts.ShowLocalOnly
+		}
+
+		selected, err := teamAppSelectPromptFunc(ctx, clients, appEnvironmentType, prompts.ShowAllApps)
```

Comment:
note: New code that prompts the developer to choose an environment (local or deployed). Then displays the proper Team Prompt. We cannot `prompts.ShowAllEnvironments` because there is guard logic to avoid having 2 apps on one team. Unfortunately, the prompt isn't aware of the app's environment, so this was our workaround.